consensus/src/lib.rs

```diff
@@ -50,11 +68,133 @@ pub enum ConsensusCreationError {
     MiningStopped,
     #[error("Mining failed")]
     MiningFailed,
+    #[error("Staking error")]
+    StakingError(#[from] ConsensusPoSError),
+    #[error("Staking failed")]
+    StakingFailed,
+    #[error("Staking stopped")]
+    StakingStopped,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub enum GenerateBlockInputData {
+    #[codec(index = 0)]
+    PoW,
+    #[codec(index = 1)]
+    PoS(Box<PoSGenerateBlockInputData>),
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+pub struct PoSGenerateBlockInputData {
+    stake_private_key: PrivateKey,
+    vrf_private_key: VRFPrivateKey,
+    pool_id: PoolId,
+    kernel_inputs: Vec<TxInput>,
+    kernel_input_utxos: Vec<TxOutput>,
+}
+
+impl PoSGenerateBlockInputData {
+    pub fn kernel_inputs(&self) -> &Vec<TxInput> {
+        &self.kernel_inputs
+    }
+
+    pub fn kernel_input_utxos(&self) -> &Vec<TxOutput> {
+        &self.kernel_input_utxos
+    }
+
+    pub fn pool_id(&self) -> PoolId {
+        self.pool_id
+    }
+
+    pub fn stake_private_key(&self) -> &PrivateKey {
+        &self.stake_private_key
+    }
+
+    pub fn stake_public_key(&self) -> PublicKey {
+        PublicKey::from_private_key(&self.stake_private_key)
+    }
+
+    pub fn vrf_private_key(&self) -> &VRFPrivateKey {
+        &self.vrf_private_key
+    }
+}
+
+#[derive(Debug, Clone, Encode, Decode)]
+pub enum FinalizeBlockInputData {
```

Comment:
Please add documentation explaining what this struct is used for. Also the one below. There's now a few structs and they can be confusing for readers.

Reply:
No problem. Done.